mlir/test/python/dialects/index_dialect.py

```diff
@@ -0,0 +1,335 @@
+# RUN: %PYTHON %s | FileCheck %s
+
+from mlir.ir import *
+from mlir.dialects import index, arith
+
+
+def run(f):
+  print("\nTEST:", f.__name__)
+  f()
+
+
+# CHECK-LABEL: TEST: testConstantOp
+@run
+def testConstantOp():
+  with Context() as ctx, Location.unknown():
+    module = Module.create()
+    with InsertionPoint(module.body):
+      a = index.ConstantOp(value=42)
+    # CHECK: %[[A:.*]] = index.constant 42
+    print(module)
+
+
+# CHECK-LABEL: TEST: testBoolConstantOp
+@run
+def testBoolConstantOp():
+  with Context() as ctx, Location.unknown():
+    module = Module.create()
+    with InsertionPoint(module.body):
+      a = index.BoolConstantOp(value=True)
+    # CHECK: %[[A:.*]] = index.bool.constant true
+    print(module)
+
+
+# CHECK-LABEL: TEST: testAndOp
+@run
+def testAndOp():
+  with Context() as ctx, Location.unknown():
+    module = Module.create()
+    with InsertionPoint(module.body):
+      a = index.ConstantOp(value=42)
+      r = index.AndOp(a, a)
+    # CHECK: %[[R:.*]] = index.and {{.*}}, {{.*}}
+    print(module)
+
+
+# CHECK-LABEL: TEST: testOrOp
+@run
+def testOrOp():
+  with Context() as ctx, Location.unknown():
+    module = Module.create()
+    with InsertionPoint(module.body):
```

Comment:
Could we lift all this module creation and printing boilerplate into the `@run` decorator?

Reply:
Surely! Thank you for the suggestion

Reply:
> Could we lift all this module creation and printing boilerplate into the `@run` decorator?

Surely! Thank you for the suggestion